dapp/src/components/AccountListener.js

```diff
@@ -10,6 +10,9 @@ import { useStoreState } from 'pullstate'
 import { setupContracts } from 'utils/contracts'
 import { login } from 'utils/account'
 
+const BALANCE_CHECK_INTERVAL = 30000 // ms
+let balanceInterval
```

Comment:
It's better to clean up during unmount than storing it as a global variable. If you return a function inside `useEffect`, it will be called when unmounting.